### PR TITLE
Improve transforms and use new LINQ-like syntax

### PR DESCRIPTION
**Pull request by Tom94**
- [x] Depends on https://github.com/ppy/osu-framework/pull/902

Adjusts the project to make use of the new LINQ-like transform chaining syntax. Right now, only very few transform usages have been migrated, and the plan is to go further step-by-step.

This PR also slightly improves the way rolling counter generic transforms work.

### CODE REVIEW

File: osu.Game/Screens/Menu/OsuLogo.cs [hunk 1]
```diff
@@ -239,21 +239,30 @@ protected override void OnNewBeat(int beatIndex, TimingControlPoint timingPoint,
             if (IsHovered)
                 this.Delay(early_activation).Schedule(() => sampleBeat.Play());
 
-            logoBeatContainer.ScaleTo(1 - 0.02f * amplitudeAdjust, early_activation, EasingTypes.Out)
-                .Then().ScaleTo(1, beatLength * 2, EasingTypes.OutQuint);
+            logoBeatContainer
+                .ScaleTo(1 - 0.02f * amplitudeAdjust, early_activation, EasingTypes.Out)
```
Comment (smoogipooo) [inline osu.Game/Screens/Menu/OsuLogo.cs]: I really dislike this being on a new line by itself. Can we use
```
logoBeatContainer.ScaleTo()
                 .Then()
                 .ScaleTo()
```
?

**Comment by Tom94** [comment]
The most recent framework improvements now also fix unintuitive overriding of transforms such as was seen in the tooltip container (this fix e.g. is no longer needed: https://github.com/ppy/osu/pull/1024 ).

As a nice side effect, some other bugs have disappeared, such as the main menu staying expanded when rapidly pressing ESC after clicking the osu logo. The following video demonstrates the fix.
https://streamable.com/a9mqf

**Comment by Tom94** [comment]
The last commit is a nice demonstration of the `OnComplete` feature of `TransformSequence`. The passed action is only invoked if the sequence completes successfully, i.e. none of its transforms gets cancelled or overwritten by another transform. In this particular case it allows the impact of the osu logo's intro sequence to only play if the transition is not interrupted; see https://streamable.com/oo0vo .

**Comment by Tom94** [comment]
Requires a disabled ReSharper warning on a purely transformed member.

**Comment by peppy** [comment]
Fomatting looks good. Also resharper's reformat code handles it all correctly, so no qualms here 👍

**Review by peppy — APPROVED** [review]

**Pull request merged**